### ISSUE TITLE
add querybuilder auto-join support for `update` queries

### ISSUE THREAD

**Issue by ogrodnek** [mikro-orm/mikro-orm#319]
**Is your feature request related to a problem? Please describe.**

The querybuilder auto-join support is really great!  Trying to run as part of select works fine, i.e.

`qb.select("*").where({activity: {enabled: true}})`

`select "e0".* from "tasks" as "e0" left join "activities" as "e1" on "e0"."activity_id" = "e1"."id" where "e1"."enabled" = $1`

but changing to use `update`, i.e.

`qb.update({someField: someValue})`

produces an error:

```
    TypeError: Cannot read property 'slice' of undefined
at ObjectCriteriaNode.autoJoin (node_modules/mikro-orm/dist/query/CriteriaNode.js:205:37)
      at ObjectCriteriaNode.process (node_modules/mikro-orm/dist/query/CriteriaNode.js:159:26)
      at node_modules/mikro-orm/dist/query/CriteriaNode.js:163:39
          at Array.reduce (<anonymous>)
      at ObjectCriteriaNode.process (node_modules/mikro-orm/dist/query/CriteriaNode.js:161:42)
      at QueryBuilder.where (node_modules/mikro-orm/dist/query/QueryBuilder.js:83:99)
```

**Describe the solution you'd like**

The same auto-join support for update queries.

**Describe alternatives you've considered**

Going to try the hand-rolled joins next

**Comment by B4nan** [comment]
I will need to investigate a bit, the thing is that with update query you can't use aliases like this, also this is not natively supported in knex (which is used to create and run the query). 

But it could be possible to do this via subquery...

**Comment by B4nan** [comment]
Closing as implemented via #537, which is now merged in v4 (`dev` branch). Subscribe here for updates #527.
